x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -432,7 +439,22 @@ public static RoleDescriptor kibanaSystemRoleDescriptor(String name) {
                 // Endpoint metrics. Kibana requires read access to send telemetry
                 RoleDescriptor.IndicesPrivileges.builder()
                     .indices("metrics-endpoint.metrics-*")
-                    .privileges("read").build()
+                    .privileges("read").build(),
+                // Fleet package upgrade
+                RoleDescriptor.IndicesPrivileges.builder()
+                    .indices("logs-*", "synthetics-*", "traces-*",
+                        "/metrics-.*&~(metrics-endpoint\\..*)/")
+                    .privileges("create_index", UpdateSettingsAction.NAME, PutMappingAction.NAME, RolloverAction.NAME)
+                    .build(),
+                // For src/dest indices of the Endpoint package that ships a transform
+                RoleDescriptor.IndicesPrivileges.builder()
+                    .indices("metrics-endpoint.metadata*")
+                    .privileges("read", "view_index_metadata")
+                    .build(),
+                RoleDescriptor.IndicesPrivileges.builder()
+                    .indices("metrics-endpoint.metadata_current_default", "metrics-endpoint.metadata_united_default")
+                    .privileges("read", "index")
```

Comment:
@joshdover The [transform API doc](https://www.elastic.co/guide/en/elasticsearch/reference/master/put-transform.html) suggests `create_index` privilege for the destination index. Just wanna double confirm it is not needed here.